### ISSUE TITLE
[a11y]: ComboBox validation error is not read out by JAWS

### ISSUE THREAD

**Issue by andreas-schmitt-mi** [carbon-design-system/carbon#10050]
### Package

carbon-components, carbon-components-react

### Browser

Firefox

### Operating System

Windows

### Package version

v10.47.0, v7.47.1

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker - Latest Deployment

### Assistive technology

JAWS 2021.2103.174

### Description

JAWS screen reader tool is not able to read the validation error of a `<ComboBox>` component. Found during manual AVT Testing. The interesting thing is, that the validation error is read out correctly for a `<Select>` component. 

Looking at the generated markup I noticed that the working `<Select>` component creates an **aria-describedby** attribute which refers to the validation error message (see screenshot below). The `<ComboBox>` component does not create such an attribute, maybe this is why JAWS does not read the error.

![image](https://user-images.githubusercontent.com/42237709/140304915-5af707db-b91b-43b0-88ec-f1a003ae7bda.png)

Just added the **aria-describedby** attribute temporarily to the `<input>` of the ComboBox and JAWS reads the validation error successfully (see screenshot below).  So I think using **aria-describedby** attribute in the same way as in the `<Select>` component should fix this issue.

![image](https://user-images.githubusercontent.com/42237709/140306325-66e50a57-6f5e-4dce-b8b9-250cae9ac226.png)

### CodeSandbox example

https://codesandbox.io/s/bold-babbage-ckktp?file=/src/index.js

### Steps to reproduce

1. Open CodeSandbox.
2. Open JAWS and navigate to the "Select language" **ComboBox** in the sandbox.
3. JAWS does not read out the validation error "Invalid selection".
4. Navigate to the "English" **Select** component.
5. JAWS does read out the validation error.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
